nilearn/decomposition/base.py

```diff
@@ -0,0 +1,533 @@
+            else:
+                raise ValueError("Object has no components_ attribute. "
+                                 "This is probably because fit has not "
+                                 "been called.")
+
+    def transform(self, imgs, confounds=None):
+        """Project the data into a reduced representation
+
+        Parameters
+        ----------
+        imgs: iterable of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data to be projected
+
+        confounds: CSV file path or 2D matrix
+            This parameter is passed to nilearn.signal.clean. Please see the
+            related documentation for details
+
+        Returns
+        ----------
+        loadings: list of 2D ndarray,
+            For each subject, each sample, loadings for each decomposition
+            components
+            shape: number of subjects * (number of scans, number of regions)
+        """
+
+        self._check_components_()
+        components_img_ = self.masker_.inverse_transform(self.components_)
+        nifti_maps_masker = NiftiMapsMasker(
+            components_img_, self.masker_.mask_img_,
+            resampling_target='maps')
+        nifti_maps_masker.fit()
+        # XXX: dealing properly with 4D/ list of 4D data?
+        if confounds is None:
+            confounds = [None] * len(imgs)
+        return [nifti_maps_masker.transform(img, confounds=confound)
+                for img, confound in zip(imgs, confounds)]
+
+    def inverse_transform(self, loadings):
+        """Use provided loadings to compute corresponding linear component
+         combination in whole-brain voxel space
+
+        Parameters
+        ----------
+        loadings: list of numpy array (n_samples x n_components)
+            Component signals to tranform back into voxel signals
+
+        Returns
+        ----------
+        reconstructed_imgs: list of nibabel.Nifti1Image
+           For each loading, reconstructed Nifti1Image
+        """
+        if not hasattr(self, 'components_'):
+            ValueError('Object has no components_ attribute. This is either '
+                       'because fit has not been called or because'
+                       '_DecompositionEstimator has direcly been used')
+        self._check_components_()
+        components_img_ = self.masker_.inverse_transform(self.components_)
+        nifti_maps_masker = NiftiMapsMasker(
+            components_img_, self.masker_.mask_img_,
+            resampling_target='maps')
+        nifti_maps_masker.fit()
+        # XXX: dealing properly with 2D/ list of 2D data?
+        return [nifti_maps_masker.inverse_transform(loading)
+                for loading in loadings]
+
+    def _raw_score(self, data, per_component=True):
+        return explained_variance(data, self.components_,
+                                  per_component=per_component)
+
+    def score(self, imgs, confounds=None, per_component=True):
```

Comment:
This function cannot be named 'score', because is the scikit-learn API, score returns only one number.